### ISSUE TITLE
Time to wait grows exponentially on large installations.

### ISSUE THREAD

**Issue by jorgeuos** [piwik/piwik#17902]
Hi,

Is there a way to get a better understanding of what the `$timeToWait` actually does?
On a large installation with 50M pageviews and a lot of segments, which some of them uses the contains option from the segment builder in the UI, the archiving process grows and spends a lot of time in `usleep` function and the archiving process can take up to 20h.

https://github.com/matomo-org/matomo/blob/0829d6e6ac487bf47e4197123d6683b271db8266/core/CliMulti.php#L418

Br, Jorge

**Comment by tsteur** [comment]
Hi @jorgeuos it's there to reduce system resources and not check constantly if a process has already finished. 

If an archiving process has been going on eg for 4 hours then it would eventually only check every 20seconds if a job has finished. This is per group of archiving jobs and not overall over all archiving jobs. So assuming that one archiving job takes eg 4hours, then the 20 seconds shouldn't have much of an impact. Or if the archiving takes 30 minutes for one archive, then a 3s wait time won't really make a difference. We could technically limit it maybe to max 5 seconds or so but I doubt it'll make a huge impact and by the looks it doesn't grow exponentially? (at least not due to the wait timeout AFAIK) It only waits basically while in the background the other command is archiving the data.

Basically, Matomo will launch a command in the background, then `$timeToWait` will define how long to wait until checking again whether the archiving command in the background has finished.

Hope this answers your question @jorgeuos

**Comment by jorgeuos** [comment]
Ok, thanks.

So you would say it's safe to have a maximum of 5 seconds. That's good to know, we have a patch for setting a fixed limit now. But then I know.

Br, Jorge
